File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.CSharp/DiagnosticsSources/ReceivedLikeUsedInReceivedInOrderDiagnosticSource.cs
using NSubstitute.Analyzers.Benchmarks.Source.CSharp.Models;

namespace NSubstitute.Analyzers.Benchmarks.Source.CSharp.DiagnosticsSources
{
    public class ReceivedLikeUsedInReceivedInOrderDiagnosticSource
    {
        public void NS5001_ReceivedLikeUsedInReceivedInOrderCallback()
        {
            var substitute = Substitute.For<Foo>();
            Received.InOrder(() =>
            {
                substitute.Received().ObjectReturningMethod();
                _ = substitute.Received().InternalObjectReturningProperty;
                _ = substitute.Received()[0];

                SubstituteExtensions.Received(substitute).ObjectReturningMethod();
                _ = SubstituteExtensions.Received(substitute).InternalObjectReturningProperty;
                _ = SubstituteExtensions.Received(substitute)[0];
            });
        }
    }
}

File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.VisualBasic/DiagnosticsSources/ReceivedLikeUsedInReceivedInOrderDiagnosticSource.vb
Imports NSubstitute.Analyzers.Benchmarks.Source.VisualBasic.Models

Namespace DiagnosticsSources
    Public Class ReceivedLikeUsedInReceivedInOrderDiagnosticSource
        Public Sub NS5001_ReceivedLikeUsedInReceivedInOrderCallback()
            Dim substitute = NSubstitute.Substitute.[For](Of Foo)()
            NSubstitute.Received.InOrder(Function()
                substitute.Received().ObjectReturningMethod()
                Dim x = substitute.Received().InternalObjectReturningProperty
                Dim y = substitute.Received()(0)
                SubstituteExtensions.Received(substitute).ObjectReturningMethod()
                Dim a = SubstituteExtensions.Received(substitute).InternalObjectReturningProperty
                Dim b = SubstituteExtensions.Received(substitute)(0)
                Throw New System.Exception()
            End Function)
        End Sub
    End Class
End Namespace
